### DISCUSSION TITLE
v2.0.0 upgrade via Github Action - post kustomization fails.

### DISCUSSION THREAD

**Discussion by filipdadgar** [fluxcd/flux2#4050] [Q&A] [answered]
Hi friends!

Decided to be a daredevil and approved upgrade to 2.0.0. As expected the actual update of gotk-components.yaml went smoothly.

Then according to these steps:

```
Upgrade procedure
Upgrade Flux from v0.x to v2.0.0 either by [rerunning bootstrap](https://fluxcd.io/flux/installation/#bootstrap-upgrade) or by using the [Flux GitHub Action](https://github.com/fluxcd/flux2/tree/main/action).

To upgrade the APIs from v1beta2, make sure the new CRDs and controllers are deployed, and then change the manifests in Git:

Remove the deprecated fields from the GitRepository and Kustomization definitions.
Commit, push and reconcile the fields removal changes.
Set apiVersion: source.toolkit.fluxcd.io/v1 in the YAML files that contain GitRepository definitions.
Set apiVersion: kustomize.toolkit.fluxcd.io/v1 in the YAML files that contain Kustomization definitions.
Set apiVersion: notification.toolkit.fluxcd.io/v1 in the YAML files that contain Receiver definitions.
Commit, push and reconcile the API version changes.
```

I updated apiVersion in gotk-sync.yaml.

So far so good.

When I started updating kustomization.yaml files in different namespaces and for specific apps to v1 version:
apiVersion: kustomize.config.k8s.io/v1beta1 --> apiVersion: kustomize.config.k8s.io/v1beta1
```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
  - namespace.yaml
  - pvc.yaml
  - helm-release.yaml
```
I started to notice kustomize build failed: accumulating resources: accumulation err='accumulating resources from './flux-system': read /tmp/kustomization- errors.

I assumed it is because my structure has a file called kustomization.yaml. So I renamed these files to customization.yaml (leap of fate :)). This instead had another undesired effect.

All objects in e.g file like below were now created even if I had a # in front.
```
---
apiVersion: kustomize.config.k8s.io/v1
kind: Kustomization
resources:
  - default
#  - minecraft
#  - hb-store
  - adguard
  - blocky
  - airplay
  #- mqtt
  #- media
  #- fc
```

What is the actual upgrade procedure? Is there something obvious I am missing to be able to use the new API version without having to modify gazillions of files? :)

Thank you for reading and thanks to the devs for awesome fluxcd experience without any issues for many many months. :)

Best,
Filip

**Comment by stefanprodan** [comment, marked as answer]
`kustomize.config.k8s.io/v1beta1` is not a Flux thing, please see the [upgrade procedure](https://github.com/fluxcd/flux2/releases/tag/v2.0.0):

| Set `apiVersion: kustomize.toolkit.fluxcd.io/v1` in the YAML files that contain Flux `Kustomization` definitions.

**Reply by filipdadgar** [reply]
oh dear! That obvious! Thanks @stefanprodan <3

**Reply by ajfriesen** [reply]
Almost fell into this as well. Thanks for clarifying!
